packages/driver/src/cy/commands/sessions.ts

```diff
@@ -0,0 +1,822 @@
+
+      _.each(origins, (u) => {
+        const $iframe = $(`<iframe src="${`${u}/__cypress/automation/getLocalStorage`}"></iframe>`)
+
+        $iframe.appendTo($iframeContainer)
+        iframes.push($iframe)
+      })
+
+      let onPostMessage
+      const successOrigins = [] as string[]
+
+      await new Promise((resolve) => {
+        onPostMessage = ((event) => {
+          const data = event.data
+
+          if (data.type !== 'localStorage') return
+
+          const value = data.value
+
+          pushValue(event.origin, value)
+
+          successOrigins.push(event.origin)
+          if (successOrigins.length === origins.length) {
+            resolve()
+          }
+        })
+
+        specWindow.addEventListener('message', onPostMessage)
+      })
+      .catch((err) => {
+        Cypress.log({
+          name: 'warning',
+          message: `failed to set session storage data on origin(s): ${_.xor(origins, successOrigins).join(', ')}`,
+        })
+      })
+      .finally(() => {
+        specWindow.removeEventListener('message', onPostMessage)
+        $iframeContainer.remove()
+      })
+
+      return getResults()
+    },
+
+    async clearStorage () {
+      const origins = await getAllHtmlOrigins()
+
+      const originOptions = origins.map((v) => ({ origin: v, clear: true }))
+
+      await sessions.setStorage({
+        localStorage: originOptions,
+        sessionStorage: originOptions,
+      })
+    },
+
+    async setStorage (options: any, clearAll = false) {
+      const current_origin = $Location.create(window.location.href).origin as string
+
+      const mapToCurrentOrigin = (v) => ({ ...v, origin: (v.origin && v.origin !== 'current_origin') ? $Location.create(v.origin).origin : current_origin })
+
+      const mappedLocalStorage = _.map(options.localStorage, (v) => {
+        const mapped = { origin: v.origin, localStorage: _.pick(v, 'value', 'clear') }
+
+        if (clearAll) {
+          mapped.localStorage.clear = true
+        }
+
+        return mapped
+      }).map(mapToCurrentOrigin)
+
+      const mappedSessionStorage = _.map(options.sessionStorage, (v) => {
+        const mapped = { origin: v.origin, sessionStorage: _.pick(v, 'value', 'clear') }
+
+        if (clearAll) {
+          mapped.sessionStorage.clear = true
+        }
+
+        return mapped
+      }).map(mapToCurrentOrigin)
+
+      const storageOptions = _.map(_.groupBy(mappedLocalStorage.concat(mappedSessionStorage), 'origin'), (v) => _.merge({}, ...v))
+
+      await _setStorageOnOrigins(storageOptions)
+    },
+
+    registerSessionHooks () {
+      Cypress.on('test:before:run:async', () => {
+        if (Cypress.config('experimentalSessionSupport')) {
+          currentTestRegisteredSessions.clear()
+
+          return navigateAboutBlank(false)
+          .then(() => sessions.clearCurrentSessionData())
+          .then(() => {
+            return Cypress.backend('reset:renderedHTMLOrigins')
```

Comment:
love the mix of caterpillar-case and camel-case

```suggestion
            return Cypress.backend('reset:rendered:html:origins')
```

Reply:
fixed